File: Dante1/Module1.06/Zadanie6.07/main.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

int main()
{
    srand(time(NULL));

    int A[4][4], B[4][4], C[4][4];


    for(int i = 0; i < 4; i++)
    {
        for(int j = 0; j < 4; j++)
        {
            A[i][j] = rand() % 10;
            printf("%2i ",A[i][j]);
        }
        printf("\n");
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        for(int j = 0; j < 4; j++)
        {
            B[i][j] = rand() % 10;
            printf("%2i ",B[i][j]);
        }
        printf("\n");
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        for(int j = 0; j < 4; j++)
        {
            if(i == j)
            {
                C[i][j] = 1;
            }
            else
            {
                C[i][j] = 0;
            }
            printf("%2i ",C[i][j]);
        }
        printf("\n");
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        int suma = 0;
        for(int j = 0; j < 4; j++)
        {
            suma = suma + A[i][j];
        }
        printf("%2i ",suma);
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        int suma = 0;
        for(int j = 0; j < 4; j++)
        {
            suma = suma + A[j][i];
        }
        printf("%2i ",suma);
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        int suma = 0;
        for(int j = 0; j < 4; j++)
        {
            suma = suma + B[i][j];
        }
        printf("%2i ",suma);
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        int suma = 0;
        for(int j = 0; j < 4; j++)
        {
            suma = suma + B[j][i];
        }
        printf("%2i ",suma);
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        int suma = 0;
        for(int j = 0; j < 4; j++)
        {
            suma = suma + C[i][j];
        }
        printf("%2i ",suma);
    }

    printf("\n");

    for(int i = 0; i < 4; i++)
    {
        int suma = 0;
        for(int j = 0; j < 4; j++)
        {
            suma = suma + C[j][i];
        }
        printf("%2i ",suma);
    }

    printf("\n");

    return 0;
}

File: Dante1/Module1.06/Zadanie6.07/main.rs
use rand::random_range;

fn print(array: & [[i32; 4]; 4])
{
    for i in 0..4
    {
        for j in 0..4
        {
            print!("{:2} ", array[i][j]);
        }
        println!();
    }
    println!();
}



fn main()
{
    let mut array_a:[[i32; 4]; 4] = [[0; 4]; 4];
    let mut array_b:[[i32; 4]; 4] = [[0; 4]; 4];
    let mut array_c:[[i32; 4]; 4] = [[0; 4]; 4];
    
    for i in 0..4
    {
        for j in 0..4
        {
            let for_a: i32 = random_range(0..10);
            let for_b: i32 = random_range(0..10);
            let for_c: i32 = random_range(0..10);
            
            array_a[i][j] = for_a;
            array_b[i][j] = for_b;
            array_c[i][j] = for_c;
        }
    }

    for i in 0..4
    {
        let mut suma: i32 = 0;
        for j in 0..4
        {
            suma = suma + array_a[i][j];
        }
        print!("{:2} ", suma);
    }
    
    println!();
    
    for i in 0..4
    {
        let mut suma: i32 = 0;
        for j in 0..4
        {
            suma = suma + array_a[j][i];
        }
        print!("{:2} ", suma);
    }

    println!();

    for i in 0..4
    {
        let mut suma: i32 = 0;
        for j in 0..4
        {
            suma = suma + array_b[i][j];
        }
        print!("{:2} ", suma);
    }

    println!();

    for i in 0..4
    {
        let mut suma: i32 = 0;
        for j in 0..4
        {
            suma = suma + array_b[j][i];
        }
        print!("{:2} ", suma);
    }

    println!();

    for i in 0..4
    {
        let mut suma: i32 = 0;
        for j in 0..4
        {
            suma = suma + array_c[i][j];
        }
        print!("{:2} ", suma);
    }

    println!();

    for i in 0..4
    {
        let mut suma: i32 = 0;
        for j in 0..4
        {
            suma = suma + array_c[j][i];
        }
        print!("{:2} ", suma);
    }

    println!();
}
